Tasks/JenkinsQueueJob/util.ts

```diff
@@ -115,47 +115,219 @@ function createRootJob(queueUri: string, jobQueue: JobQueue, taskOptions: TaskOp
+            failReturnCode(httpResponse, 'crumb request failed.');
+            defer.reject(getFullErrorMessage(httpResponse, 'Crumb request failed.'));
+        } else {
+            taskOptions.crumb = body.slice(body.indexOf(':') + 1);
+            tl.debug('crumb: ' + taskOptions.crumb);
+            defer.resolve(taskOptions.crumb);
+        }
+    }).auth(taskOptions.username, taskOptions.password, true);
+    return defer.promise;
+}
+
+/**
+ * Supported parameter types: boolean, string, choice, password
+ * 
+ * - If a parameter is not defined by Jenkins it is fine to pass it anyway
+ * - Anything passed to a boolean parameter other than 'true' (case insenstive) becomes false.
+ * - Invalid choice parameters result in a 500 response.
+ * 
+ */
+function parseJobParameters(jobParameters: string[]): any {
+    var formData = {};
+    for (var i = 0; i < jobParameters.length; i++) {
+        var paramLine = jobParameters[i].trim();
+        var splitIndex = paramLine.indexOf('=');
+        if (splitIndex <= 0) { // either no paramValue (-1), or no paramName (0)
+            throw 'Job parameters should be specified as "parameterName=parameterValue" with one name, value pair per line. Invalid parameter line: ' + jobParameters[i];
+        }
+        var paramName = paramLine.substr(0, splitIndex).trim();
+        var paramValue = paramLine.slice(splitIndex + 1).trim();
+        formData[paramName] = paramValue;
+    }
+    return formData;
+}
+
+function getTeamParameters(): any {
+    var formData = {};
+    allTeamBuildVariables.forEach(variableName => {
+        let paramValue = tl.getVariable(variableName);
+        if (paramValue) {
+            let paramName = '_team-build_' + variableName;
+            formData[paramName] = paramValue;
+        }
+    });
+    return formData;
+}
+
+//https://www.visualstudio.com/docs/build/define/variables
+var allTeamBuildVariables: string[] = [
+    //control variables
+    'Build.Clean',
+    'Build.SyncSources',
+    'System.Debug',
+    //predefined variables
+    'Agent.BuildDirectory',
+    'Agent.HomeDirectory',
+    'Agent.Id',
+    'Agent.MachineName',
+    'Agent.Name',
+    'Agent.WorkFolder',
+    'Build.ArtifactStagingDirectory',
+    'Build.BuildId',
+    'Build.BuildNumber',
+    'Build.BuildUri',
+    'Build.BinariesDirectory',
+    'Build.DefinitionName',
+    'Build.DefinitionVersion',
+    'Build.QueuedBy',
+    'Build.QueuedById',
+    'Build.Repository.Clean',
+    'Build.Repository.LocalPath',
+    'Build.Repository.Name',
+    'Build.Repository.Provider',
+    'Build.Repository.Tfvc.Workspace',
+    'Build.Repository.Uri',
+    'Build.RequestedFor',
+    'Build.RequestedForId',
+    'Build.SourceBranch',
+    'Build.SourceBranchName',
+    'Build.SourcesDirectory',
+    'Build.SourceVersion',
+    'Build.StagingDirectory',
+    'Build.Repository.Git.SubmoduleCheckout',
+    'Build.SourceTfvcShelveset',
+    'Common.TestResultsDirectory',
+    'System.AccessToken',
```

Comment:
We shouldn't pass the OAuth access token to Jenkins (`System.AccessToken`).  Jenkins has its own credentials for connecting to TFS/Team Services.